MapboxNavigation/NavigationViewController.swift

```diff
@@ -20,7 +20,7 @@ public protocol NavigationViewControllerDelegate {
     /**
      Called when the user arrives at the destination.
      */
-    @objc optional func navigationViewController(_ navigationViewController : NavigationViewController, didArriveAt destination: MGLAnnotation)
+    @objc optional func navigationViewController(_ navigationViewController : NavigationViewController, didArriveAt destination: Waypoint)
```

Comment:
Should we also add `navigationViewControllerDidArrive(_:)` (with only one argument of type NavigationViewController)?